src/main/java/com/leverx/employeestat/rest/controller/DepartmentController.java

```diff
@@ -1,56 +1,83 @@
 
-@RestController("department")
+@RestController
 @RequestMapping("/api/departments")
 public class DepartmentController {
 
     private final DepartmentService departmentService;
-    private final DepartmentConverter departmentConverter;
+    private final BindingResultParser bindingResultParser;
 
     @Autowired
-    public DepartmentController(DepartmentService departmentService, DepartmentConverter departmentConverter) {
+    public DepartmentController(DepartmentService departmentService, BindingResultParser bindingResultParser) {
         this.departmentService = departmentService;
-        this.departmentConverter = departmentConverter;
+        this.bindingResultParser = bindingResultParser;
     }
 
     @GetMapping
     @ResponseStatus(HttpStatus.OK)
     public List<DepartmentDTO> getAllDepartments() {
-        return departmentService.getAll()
-                .stream()
-                .map(departmentConverter::toDTO)
-                .collect(Collectors.toList());
+        return departmentService.getAll();
+    }
+
+    @GetMapping("/{id}")
+    @ResponseStatus(HttpStatus.OK)
+    public DepartmentDTO getDepartment(@PathVariable("id") String id) {
+        return departmentService.getById(getUUIDFromString(id));
     }
 
     @PostMapping
     @ResponseStatus(HttpStatus.OK)
-    public DepartmentDTO postDepartment(@RequestBody DepartmentDTO departmentDTO) {
-        Department department = departmentConverter.toEntity(departmentDTO);
-        return departmentConverter.toDTO(departmentService.save(department));
+    public DepartmentDTO postDepartment(@RequestBody @Valid DepartmentDTO departmentDTO, BindingResult result) {
+        if (result.hasErrors()) {
+            throw new NotValidRecordException("Fields of Department have errors: " +
+                    bindingResultParser.getFieldErrMismatches(result));
+        }
+        return departmentService.save(departmentDTO);
     }
 
     @PutMapping
     @ResponseStatus(HttpStatus.OK)
-    public DepartmentDTO putDepartment(@RequestBody DepartmentDTO departmentDTO) {
-        Department department = departmentConverter.toEntity(departmentDTO);
-        return departmentConverter.toDTO(departmentService.update(department));
+    public DepartmentDTO putDepartment(@RequestBody @Valid DepartmentDTO departmentDTO, BindingResult result) {
+        if (result.hasErrors()) {
+            throw new NotValidRecordException("Fields of Department have errors: " +
+                    bindingResultParser.getFieldErrMismatches(result));
+        }
+        return departmentService.update(departmentDTO);
     }
 
     @DeleteMapping("/{id}")
     @ResponseStatus(HttpStatus.OK)
-    public void deleteDepartment(@PathVariable("id") UUID id) {
-        departmentService.deleteById(id);
+    public void deleteDepartment(@PathVariable("id") String id) {
+        departmentService.deleteById(getUUIDFromString(id));
+    }
+
+    private UUID getUUIDFromString(String id) {
```

Comment:
I think it makes sense to extract it as some static utility method in the some utils class